### PR TITLE
6.0.0 release updates

### LINKED ISSUE #2887: year  needs to be changed in welcome window
**Describe the bug**
apparently the Welcome window (which contains the splash screen) has an independent entry for the (c) date. 2023 needs updating to 2024.

**SasView version (please complete the following information):**
 - Version: 6.0.0a1

**Additional context**
Needs to be fixed as part of the 6.0 release process

### PR DESCRIPTION
**Pull request by krzywon**
## Description

This is an update to the release related information specific to 6.0.0.

Changes:
- Release notes
- Sister package release branches (sasmodels and bumps)
- DOI generated by Zenodo for v6.0.0
- Copyright year is now 2024
- Sasview version is now 6.0.0 (or 6.0.0b1 in some cases)
- Updates to `build_tools/release_automation.py`
  - Include sasdata for versioning and issues
  - Ability to read in contributor list from tsv file (if found)
  - Include help for the script
  - General code cleanup and lint

Still required:
- All versions should be 6.0.0 for the final release
- Update to contributors for the zenodo record

Fixes #2887

## Review Checklist:

**Documentation** (check at least one)
- [ ] There is **nothing** that needs documenting
- [x] Documentation changes are **in this PR**
- [ ] There is an **issue** open for the documentation (link?)

**Installers**
- [ ] There is a chance this will affect the **installers**, if so
  - [ ] **Windows** installer (GH artifact) has been tested (installed and worked) 
  - [ ] **MacOSX** installer (GH artifact) has been tested (installed and worked) 

**Licencing** (untick if necessary)
- [x] The introduced changes comply with SasView license (BSD 3-Clause)

### CODE REVIEW

File: build_tools/release_automation.py
Comment (butlerpd) [inline build_tools/release_automation.py:159]: I assume that this will mean that if both Producer and Other is checked only Other will get recorded? Probably reasonable but not sure what we agreed.. so far?

Also, could one future proof this by taking the record['type'] from the header line? not suggesting we need to make big changes now. Just asking the question.

File: build_tools/release_automation.py
Comment (butlerpd) [inline build_tools/release_automation.py:213]: This line seems superfluous and outdated?  The original concept was that the installer would have a build number (as many software packages do) which would be the commit hash, but that only worked briefly. Does "GIT_COMMIT" have any utility? Of course, even if it is pointless, it may be something to remove after the release in which case we should ticket it.

File: docs/sphinx-docs/source/user/RELEASE.rst
Comment (butlerpd) [inline docs/sphinx-docs/source/user/RELEASE.rst:148]: This should say Change GUI package from PyQt5 to Pyside6. Also, This actually a big deal right? - no doubt some breaking changes and part of why this  is 6 not 5.1?

File: src/sas/system/version.py
Comment (butlerpd) [inline src/sas/system/version.py:4]: Same comment as for `releasese_automation.py` - What is the point of `__build__` ?

File: docs/sphinx-docs/source/user/RELEASE.rst
Comment (butlerpd) [inline docs/sphinx-docs/source/user/RELEASE.rst:32]: again, lines beginning with numbers that I don't understand? Probably just need to remove them?

ALSO ... I don't think the particle editor is actually complete and available is it?

File: docs/sphinx-docs/source/user/RELEASE.rst
Comment (butlerpd) [inline docs/sphinx-docs/source/user/RELEASE.rst:19]: Going from v5 to v6 means more than feature additions. In principle breaking changes? In particular moving sasdata out of sasview and moving from PyQt to Pyside for the GUI package? Seems like it should be made clear somehow why this is not 5.1 (a feature release)?

Also - for a future PR probably but I would recommend highlighting at the top some of the big changes for users like a proper settings option and the combined orders of magnitude speed increases in both reading the PDB and then calculating the scattering from that PDB? Or maybe we just leave those now in "What's new" window?

**Review by butlerpd — COMMENTED** [review]
I think there are at least a few items that should be addressed before merging. I still need to test the installer before finishing the review

File: build_tools/release_automation.py
Comment (krzywon) [reply build_tools/release_automation.py:159]: > I assume that this will mean that if both Producer and Other is checked only Other will get recorded? Probably reasonable but not sure what we agreed.. so far?

Correct. This also does not allow names to be in both the `creators` and `contributors` list.

> Also, could one future proof this by taking the record['type'] from the header line? not suggesting we need to make big changes now. Just asking the question.

I don't see why not, but I agree this can wait.

File: build_tools/release_automation.py
Comment (krzywon) [reply build_tools/release_automation.py:213]: Good question. Probably not necessary, but I agree we shouldn't hold up a beta release for this.

**Comment by butlerpd** [comment]
The sphinx documentation does not have the correct version number. There is a separate place it gets set.

**Review by butlerpd — APPROVED** [review]
as mentioned the version of the sphinx docs needs changing. Otherwise it should be good to go. Approving assuming these things get fixed.

**Comment by krzywon** [comment]
> The sphinx documentation does not have the correct version number. There is a separate place it gets set.

I'll take a look at this tomorrow.

**Comment by krzywon** [comment]
> The sphinx documentation does not have the correct version number. There is a separate place it gets set.

Another place this is hard-coded that I missed. This build should have the correct version displayed in the documentation.

**Pull request merged**
